Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -1771,6 +1771,11 @@ theorem mk_denumerable (α : Type u) [Denumerable α] : #α = ℵ₀ :=
   denumerable_iff.1 ⟨‹_›⟩
 #align cardinal.mk_denumerable Cardinal.mk_denumerable
 
+theorem _root_.Set.countable_infinite_iff_nonempty_denumerable {α : Type*} {s : Set α} :
```

Comment:
We have https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Denumerable.html#nonempty_denumerable_iff
You can use it here (or migrate SEP to use it instead).

Reply:
Thanks; I didn't see this. I have golfed this proof accordingly.

I'm open to migrating sphere-eversion to nonempty_denumerable_iff instead (trying locally, that seems doable). Do you think this lemma is worth having anyway?